x-pack/plugins/ingest/public/secops/pages/home/index.tsx

```diff
@@ -0,0 +1,19 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React from 'react';
+
+import { ColumnarPage } from '../../../components/page';
+
+export class HomePage extends React.PureComponent {
+  public render() {
+    return (
+      <ColumnarPage>
+        <h1>Hello SecOps</h1>
+      </ColumnarPage>
+    );
+  }
+}
```

Comment:
Nothing wrong with this PureComponent, but I do want to bring up the HOC alternative approach typically called `purification or purify or pure HOC` with fewer characters since I see Kibana includes the library `recompose` already.


Someone forgot to add types for it at some point, so you do that first

```sh
yarn add @types/recompose
```

Then you can write your pure components like so:

 ```ts
import React from 'react';
import { pure } from 'recompose';

import { ColumnarPage } from '../../../components/page';

export const HomePage = pure(() => (
  <ColumnarPage>
    <h1>Hello SecOps</h1>
  </ColumnarPage>
));
```

Double check my work though if you want to go this way vs. creating classes for PureComp's.
https://github.com/acdlite/recompose/blob/master/docs/API.md#pure
https://vasanthk.gitbooks.io/react-bits/perf-tips/02.pure-component.html

Reply:
but I am ok do it this way

Reply:
because both use shouldComponentUpdate to shallowly detect changes, so there's no practical difference between them, as long as a component doesn't involve local state.